pkg/diagnostic/diagnostic.go

```diff
@@ -0,0 +1,45 @@
+package diagnostic
+
+// Type is a type of diagnostic.
+type Type int
+
+const (
+	// TypeUnknown designates an unknown type of diagnostic.
+	TypeUnknown Type = iota
+	// TypeLogs designates a Diagnostic for logs.
+	TypeLogs
+	// TypeStack designates a Diagnostic for stack trace.
+	TypeStack
+	// TypeTLS designates a Diagnostic for tls logs.
+	TypeTLS
```

Comment:
Is that a preparation for something? It's only used for testing purposes.

Reply:
Yes, it is the preparation for sending TLS logs. Do you think we should remove it from this PR?

Reply:
If any error during the commands execution will be caught, do we really need this type? Aren't `logs` and `stack` enough? As far I understood this type will be for specific SSL errors. Can't we always send them as "generic"?

Reply:
@gandarez To be honest, I did not find a way yet to retrieve these tls logs. I was playing around with capturing stdout and stderr, but this is a little tricky and also no tls logs showed up. Even when I set `http1debug=1` via `GODEBUG`. I will remove `TypeTLS` and then we can have a discussion on it outside of this PR. :+1: